src/main/java/org/spongepowered/api/entity/living/player/tab/TabList.java

```diff
@@ -70,45 +72,46 @@
      * @param header The new header
      * @param footer The new footer
      */
-    default void setHeaderAndFooter(Text header, Text footer) {
+    default void setHeaderAndFooter(@Nullable Text header, @Nullable Text footer) {
         this.setHeader(header);
         this.setFooter(footer);
     }
 
     /**
-     * Gets the players on the list. The list should be immutable.
+     * Gets the entries on the list.
+     *
+     * <p>The returned collection should be immutable.</p>
      *
-     * @return The players on the list
+     * @return The entries on the list
      */
-    List<PlayerTabInfo> getPlayers();
+    Collection<TabListEntry> getEntries();
 
     /**
-     * Adds a player to the list.
+     * Gets a {@link TabListEntry} matching the specified unique id.
      *
-     * @param player The player to add
-     * @throws IllegalArgumentException when it attempts to add a player already
-     *         on the list. This is to prevent modification of a
-     *         {@link PlayerTabInfo} by overwriting it
+     * @param uniqueId The unique id to search for
+     * @return The entry if present, otherwise {@link Optional#empty()}
      */
-    void addPlayer(PlayerTabInfo player) throws IllegalArgumentException;
+    Optional<TabListEntry> getEntry(UUID uniqueId);
 
     /**
-     * Removes a player from the list. This should only be used to completely
-     * remove a player, not add it back later. Note that if this is used on a
-     * player, but they remain visible in-game, their skin will not work.
+     * Adds an entry to the list.
      *
-     * @param playerId the UUID of the player to remove
-     * @return The {@link PlayerTabInfo} that was associated with the UUID
+     * @param builder The entry to add
+     * @throws IllegalArgumentException if an entry already with the same unique
+     *     id exists on the list
      */
-    PlayerTabInfo removePlayer(UUID playerId);
+    void addEntry(TabListEntry.Builder builder) throws IllegalArgumentException;
```

Comment:
Isn't that what the `build()` method is for?
If not because you need more data than you should not expose the build method at all or is there a different usecase for it?